torax/_src/transport_model/tglf_based_transport_model.py

```diff
@@ -332,4 +338,62 @@ def _prepare_tglf_inputs(
         delta_shear=delta_shear,
         beta_e=beta_e,
         Zeff=core_profiles.Z_eff_face,
+        Q_GB=Q_GB,
+        Gamma_GB=Gamma_GB,
+    )
+
+  @override
+  def _make_core_transport(
+      self,
+      electron_heat_flux_GB: jax.Array,
+      ion_heat_flux_GB: jax.Array,
+      electron_particle_flux_GB: jax.Array,
+      tglf_inputs: TGLFInputs,
+      transport: RuntimeParams,
+      geo: geometry.Geometry,
+      core_profiles: state.CoreProfiles,
+  ) -> transport_model_lib.TurbulentTransport:
+    # Denormalised TGLF output fluxes
+    Q_e = electron_heat_flux_GB * tglf_inputs.Q_GB  # [W/m^2]
+    Q_i = ion_heat_flux_GB * tglf_inputs.Q_GB  # [W/m^2]
+    Gamma_e = electron_particle_flux_GB * tglf_inputs.Gamma_GB  # [s^-1/m^2]
+
+    # Total thermal power and particle rate
+    dV_drho = geo.vpr_face / geo.rho_b
+    P_e = Q_e * dV_drho  # [W]
+    P_i = Q_i * dV_drho  # [W]
+    S_e = Gamma_e * dV_drho  # [s^-1]
+
+    # Convert from power to chi
+    # Note: g1/vpr = ⟨(∇ρₙ)²⟩ ∂V/∂ρₙ, and has units [m]
+    dT_e_drhon = core_profiles.T_e.face_grad() * constants.CONSTANTS.keV_to_J
+    dT_i_drhon = core_profiles.T_i.face_grad() * constants.CONSTANTS.keV_to_J
+    chi_e = -P_e / (
+        core_profiles.n_e.face_value() * dT_e_drhon * geo.g1_over_vpr_face
+    )
+    chi_i = -P_i / (
+        core_profiles.n_i.face_value() * dT_i_drhon * geo.g1_over_vpr_face
+    )
```

Comment:
The denormalisation of TGLF is slightly different to QuaLiKiz.

In particular, TGLF normalises everything with respect to the electron temperature and density, which means that chi_i can't be denormalised in the same way.

Due to the difficulties I had trying to track down bugs in this interface, I would strongly advocate for this sort of layout - denormalisation, conversion to power, conversion to chi - rather than what is done in QLKNN. 

However, if there are any speed improvements that can be made by changing what is computed, they definitely take priority.